algorithms.py

```diff
@@ -1186,12 +1204,23 @@ def migration_event(self, j, k):
             u = u.next
         # print("AFTER Population sizes:", [len(pop) for pop in self.P])
 
-    def increment_segment_mass(self, seg, left, right):
-        mass = self.recomb_map.mass_between(left, right)
+    def set_segment_left_endpoint(self, seg, pos):
+        seg.left = pos
+        seg.left_mass = self.recomb_map.position_to_mass(pos)
+
+    def increment_segment_mass_between(self, seg, l_mass, r_mass):
+        self.L[seg.label].increment(seg.index, r_mass - l_mass)
+
+    def increment_segment_mass(self, seg, seg2):
```

Comment:
Maybe ``add_segment_mass(seg1, seg2)`` with a comment saying "adds the mass subtended between the endpoints of seg2 to seg1" or something.